stripecardscan/src/main/java/com/stripe/android/stripecardscan/framework/util/AcceptedImageConfigs.kt

```diff
@@ -13,77 +13,70 @@ internal enum class ImageFormat(val string: String) {
     companion object {
         fun fromValue(value: CardImageVerificationDetailsFormat): ImageFormat =
             when (value) {
-                CardImageVerificationDetailsFormat.HEIC -> ImageFormat.HEIC
-                CardImageVerificationDetailsFormat.JPEG -> ImageFormat.JPEG
-                CardImageVerificationDetailsFormat.WEBP -> ImageFormat.WEBP
+                CardImageVerificationDetailsFormat.HEIC -> HEIC
+                CardImageVerificationDetailsFormat.JPEG -> JPEG
+                CardImageVerificationDetailsFormat.WEBP -> WEBP
             }
     }
 }
 
+internal data class OptionalImageSettings(
+    val compressionRatio: Double?,
+    val imageSize: Size?,
+    val imageCount: Int?
+) {
+    constructor(settings: CardImageVerificationDetailsImageSettings?) : this(
+        compressionRatio = settings?.compressionRatio,
+        imageSize = settings?.imageSize?.takeIf { it.size > 1 }?.let {
+            Size(it.first().toInt(), it.last().toInt())
+        },
+        imageCount = settings?.imageCount
+    )
+}
+
 internal data class ImageSettings(
-    var compressionRatio: Double? = null,
-    var imageSize: Size? = null
+    val compressionRatio: Double,
+    val imageSize: Size,
+    val imageCount: Int
 ) {
     companion object {
         // These default values are what Android was using before the addition of a server config.
-        val DEFAULT = ImageSettings(0.92, Size(1080, 1920))
-    }
-
-    constructor(settings: CardImageVerificationDetailsImageSettings?) : this() {
-        settings?.let {
-            compressionRatio = it.compressionRatio ?: compressionRatio
-
-            it.imageSize?.takeIf { it.size > 1 }?.let {
-                var pendingSize = imageSize ?: ImageSettings.DEFAULT.imageSize!!
-                val width = it.first().toInt() ?: pendingSize.width
-                val height = it.last().toInt() ?: pendingSize.height
-                imageSize = Size(width, height)
-            }
-        }
+        val DEFAULT = ImageSettings(0.92, Size(1080, 1920), 3)
```

Comment:
@jaimepark-stripe these are the default values

Reply:
oh woops! didnt see this. ty!